docs/languages/csharp/expression-trees-interpreting.md

```diff
@@ -0,0 +1,569 @@
+# Interpreting Expressions
+
+Now, let's write some code to examine the structure of an 
+*expression tree*. Every node in an expression tree will be
+an object of a class that is derived from `Expression`.
+
+That design makes visiting all the nodes in an expression tree
+a relatively straight forward recursive operation. The general strategy
+is to start at the root node and determine what kind of node it is.
+
+If the node type has children, recursively visit the children. At each
+child node, repeat the process used at the root node: determine the
+type, and if the type has children, visit each of the children.
+
+## Examining an Expression with No Children
+Let's start by visiting each node in a very simple expression tree.
+Here's the code that creates a constant expression and then
+examines its properties:
+
+```cs
+var constant = Expression.Constant(24, typeof(int));
+
+Console.WriteLine($"This is an {constant.NodeType} expression type");
```

Comment:
Should this sentence read as "This is a/an {" since you don't know what NodeType value you can get back?